### PR TITLE
Change what is accepted by ModelRc::new and ModelRc::from

### PR DESCRIPTION
**Pull request by ogoffart**
- ModelRc::new constructs a ModelRc from a impl Model
 - ModelRc::form constructs a ModelRc from a `Rc<dyn Model>` or `Rc<impl Model>`

### CODE REVIEW

**Comment by ogoffart** [comment]
I think this makes the API easier. What do you think?

File: docs/tutorial/rust/src/main_game_logic_in_rust.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ fn main() {
     // ANCHOR: game_logic
     // Assign the shuffled Vec to the model property
     let tiles_model = std::rc::Rc::new(sixtyfps::VecModel::from(tiles));
-    main_window.set_memory_tiles(sixtyfps::ModelRc::new(tiles_model.clone()));
+    main_window.set_memory_tiles(sixtyfps::ModelRc::from(tiles_model.clone()));
```
Comment (tronical) [inline docs/tutorial/rust/src/main_game_logic_in_rust.rs:23]: Is this also possible?

```suggestion
    main_window.set_memory_tiles(tiles_model.clone().into());
```

File: api/sixtyfps-rs/migration.md [hunk 2]
```diff
@@ -62,6 +62,10 @@ fn model_tracker(&self) -> &dyn ModelTracker {
...
 ```
 
-#### Minor changes to `Model`
+#### ModelHandle
 
 `ModelHandle` was renamed [`ModelRc`].
+
+[`ModelRc::new`]  no longer takes a `Rc`, but takes the structure that implements model directly.
+To construct a `ModelRc` from a Rc, one should use the `From` trait. So [`ModelRc::from`] is doing what
+`ModelHandle::new` was doing.
```
Comment (tronical) [inline api/sixtyfps-rs/migration.md:71]:
```suggestion
[`ModelRc::new`]  no longer takes a `Rc`, but takes the structure that implements the [`Model`] trait directly.
To construct a `ModelRc` from a Rc for your model, use the `From` trait. [`ModelRc::from`] is doing what
`ModelHandle::new` was doing.
```

File: sixtyfps_runtime/corelib/model.rs [hunk 3]
```diff
@@ -462,8 +461,17 @@ impl Model for bool {
     }
 }
 
-/// Properties of type Array in the .60 language are represented as
-/// a `ModelRc` that implements the [`Model`] trait.
+/// A Reference counted [`Model`].
+///
+/// The `ModelRc` struct holds something that impelement the [`Model`] trait.
```
Comment (tronical) [inline sixtyfps_runtime/corelib/model.rs:466]:
```suggestion
/// The `ModelRc` struct holds something that implements the [`Model`] trait.
```

File: sixtyfps_runtime/corelib/model.rs [hunk 4]
```diff
@@ -462,8 +461,17 @@ impl Model for bool {
...
+/// This is used in `for` expression in the .60 language.
```
Comment (tronical) [inline sixtyfps_runtime/corelib/model.rs:467]:
```suggestion
/// This is used in `for` expressions in the .60 language.
```

File: sixtyfps_runtime/corelib/model.rs [hunk 5]
```diff
@@ -462,8 +461,17 @@ impl Model for bool {
...
+/// Array properties sin the .60 language are actually holding a ModelRc.
```
Comment (tronical) [inline sixtyfps_runtime/corelib/model.rs:468]:
```suggestion
/// Array properties in the .60 language are holding a ModelRc.
```

File: docs/tutorial/rust/src/main_game_logic_in_rust.rs (hunk 1 above)
Comment (ogoffart) [reply docs/tutorial/rust/src/main_game_logic_in_rust.rs:23]: yes, i'll change that.

File: sixtyfps_runtime/corelib/model.rs [hunk 6]
```diff
@@ -498,11 +507,31 @@ impl<T> core::cmp::PartialEq for ModelRc<T> {
 }
 
 impl<T> ModelRc<T> {
-    pub fn new(model: Rc<dyn Model<Data = T>>) -> Self {
+    pub fn new(model: impl Model<Data = T> + 'static) -> Self {
+        Self(Some(Rc::new(model)))
+    }
+}
+
+impl<T, M: Model<Data = T> + 'static> From<Rc<M>> for ModelRc<T> {
```
Comment (tronical) [inline sixtyfps_runtime/corelib/model.rs:515]: `From<&Rc<M>>` would be another variant we could offer. Not sure if worth it though.

**Review by tronical — APPROVED** [review]
I think that's an improvement.

File: sixtyfps_runtime/corelib/model.rs (hunk 6 above)
Comment (ogoffart) [reply sixtyfps_runtime/corelib/model.rs:515]: We can always add that later if we need.

**Pull request merged**